kong-dbless.yaml

```diff
@@ -0,0 +1,109 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  namespace: kong
+  name: kong-dbless
+  labels:
+    app: kong-dbless
+spec:
+  selector:
+    matchLabels:
+      app: kong-dbless
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+    type: RollingUpdate
+  template:
+    metadata:
+      annotations:
+        k8s.konghq.com/sidecar-inject: "false"
+        prometheus.io/port: "8001"
+        prometheus.io/scrape: "true"
+      labels:
+        app: kong-dbless
+    spec:
+      containers:
+      - name: kong-dbless
+        image: kong
+        imagePullPolicy: IfNotPresent
+        env:
+          - name: KONG_DATABASE
+            value: 'off'
+          - name: KONG_LOG_LEVEL
+            value: debug
+          - name: KONG_ADMIN_ACCESS_LOG
+            value: /dev/stdout
+          - name: KONG_PROXY_ERROR_LOG
+            value: /dev/stderr
+          - name: KONG_ADMIN_ERROR_LOG
+            value: /dev/stderr
+          - name: KONG_ADMIN_LISTEN
+            value: 0.0.0.0:8001,0.0.0.0:8444 ssl
+          - name: KONG_PROXY_LISTEN
+            value: 0.0.0.0:8000,0.0.0.0:8443 ssl
+          - name: KONG_DECLARATIVE_CONFIG
+            value: /etc/kong/declarative.yaml
+        volumeMounts:
+          - name: kongdeclarative
+            mountPath: /etc/kong/declarative.yaml
+            subPath: declarative.yaml
+        ports:
+        - name: data-http
+          containerPort: 8000
+        - name: data-https
+          containerPort: 8443
+        - name: admin-http
+          containerPort: 8001
+        - name: admin-https
+          containerPort: 8444
+        livenessProbe:
+          failureThreshold: 3
+          httpGet:
+            path: /status
+            port: 8001
+            scheme: HTTP
+          initialDelaySeconds: 30
+          periodSeconds: 10
+          successThreshold: 1
+          timeoutSeconds: 1
+        readinessProbe:
+          failureThreshold: 3
+          httpGet:
+            path: /status
+            port: 8001
+            scheme: HTTP
+          periodSeconds: 10
+          successThreshold: 1
+          timeoutSeconds: 1
+      volumes:
+      - name: kongdeclarative
+        configMap:
+          name: kongdeclarative
+---
+apiVersion: v1
+kind: Service
+metadata:
+  namespace: kong
+  name: kong-dbless
+spec:
+  type: NodePort
+  ports:
+  - name: kong-proxy-http
+    port: 8000
+    targetPort: 8000
+    protocol: TCP
+  - name: kong-proxy-https
+    port: 8443
+    targetPort: 8443
+    protocol: TCP
+  - name: http
+    port: 8001
+    targetPort: 8001
+    protocol: TCP
+  - name: https
+    port: 8444
+    targetPort: 8444
+    protocol: TCP
```

Comment:
This is overloading the service with multiple concerns.
We should split this into two k8s services: kong-proxy and kong-admin.

Reply:
I'm a little hazy as to what the separation of concerns should look like with dbless / declarative.

> We should split this into two k8s services: kong-proxy and kong-admin.
Counterpoint: dbless shouldn't have an admin api in which case should it just be turned off entirely?

Reply:
discussed offline and agreed dbless shouldn't expose admin api at all 👍